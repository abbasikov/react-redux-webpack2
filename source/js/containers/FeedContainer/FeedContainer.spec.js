import React from 'react';
import {shallow, render} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import FeedContainer from './FeedContainer';
import toJson from 'enzyme-to-json';
import { INITIAL_STATE } from '../../constants/AppConstants';
import sinon from 'sinon';

const mockStore = configureMockStore([ thunk ]);

describe('FeedContainer', function() {
    var store;
    var container;

    beforeEach(function() {
        store = mockStore(INITIAL_STATE);
        container = render(<Provider store={store}>
                                <FeedContainer />
                            </Provider>);
    });

    describe('#renders', () => {

        it('with initial state, when feeds are empty', function() {
            expect(toJson(container)).toMatchSnapshot();
        });
    });
});