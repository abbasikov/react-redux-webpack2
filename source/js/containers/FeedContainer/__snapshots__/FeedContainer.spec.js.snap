// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedContainer #renders with initial state, when feeds are empty 1`] = `
<div
  class="content-wrapper"
>
  <div
    class="panel panel-flat "
  >
    <div
      class="panel-body panel-post"
    >
      <button
        class="btn bg-orange-600 btn-xs"
        type="button"
      >
        <i
          class="glyphicon glyphicon-plus position-left"
        />
        <span>
          POST
        </span>
      </button>
    </div>
  </div>
  <div
    class="l1-core-rail"
  >
    <div
      class="panel post-feed delay260"
    >
      <div
        class="circle"
      />
      <div
        class="line-wrapper"
      >
        <div
          class="line width110"
        />
        <div
          class="line width250"
        />
      </div>
      <div
        class="height20"
      />
      <div
        class="line width500"
      />
      <div
        class="line width470"
      />
    </div>
    <div
      class="panel post-feed delay320"
    >
      <div
        class="circle"
      />
      <div
        class="line-wrapper"
      >
        <div
          class="line width110"
        />
        <div
          class="line width250"
        />
      </div>
      <div
        class="height20"
      />
      <div
        class="line width500"
      />
      <div
        class="line width470"
      />
    </div>
    <div
      class="panel post-feed delay380"
    >
      <div
        class="circle"
      />
      <div
        class="line-wrapper"
      >
        <div
          class="line width110"
        />
        <div
          class="line width250"
        />
      </div>
      <div
        class="height20"
      />
      <div
        class="line width500"
      />
      <div
        class="line width470"
      />
    </div>
  </div>
</div>
`;
